nifi-toolkit/nifi-toolkit-cli/src/main/java/org/apache/nifi/toolkit/cli/impl/command/registry/flow/ImportAllFlows.java

```diff
@@ -0,0 +1,330 @@
+
+        // As we need to keep the version order the list needs to be sorted
+        files.sort((o1, o2) -> ComparisonChain.start()
+                .compare(o1.getBucketName(), o2.getBucketName())
+                .compare(o1.getFlowName(), o2.getFlowName())
+                .compare(o1.getVersion(), o2.getVersion())
+                .result());
+
+        for (VersionFileMetaData file : files) {
+            final String inputSource = file.getInputSource();
+            final String fileContent = getInputSourceContent(inputSource);
+            final VersionedFlowSnapshot snapshot = MAPPER.readValue(fileContent, VersionedFlowSnapshot.class);
+
+            final String bucketName = snapshot.getBucket().getName();
+            final String bucketDescription = snapshot.getBucket().getDescription();
+            final String flowName = snapshot.getFlow().getName();
+            final String flowDescription = snapshot.getFlow().getDescription();
+            final int flowVersion = snapshot.getSnapshotMetadata().getVersion();
+            // The original bucket and flow ids must be kept otherwise NiFi won't be able to synchronize with the NiFi Registry
+            final String flowId = snapshot.getFlow().getIdentifier();
+            final String bucketId = snapshot.getBucket().getIdentifier();
+
+            // Create bucket if missing
+            if (bucketMap.containsKey(bucketName)) {
+                printMessage(isInteractive, bucketName + SKIPPING_BUCKET_CREATION);
+            } else {
+                //The original bucket id must be kept otherwise NiFi won't be able to synchronize with the NiFi Registry
+                createBucket(client, bucketMap, bucketName, bucketDescription, bucketId);
+            }
+
+            // Create flow if missing
+            if (flowMap.containsKey(new ImmutablePair<>(bucketId, flowName))) {
+                if (skip) {
+                    printMessage(isInteractive, flowName + SKIPPING_IMPORT);
+                    continue;
+                } else {
+                    //flowId
+                    printMessage(isInteractive, flowName + SKIPPING_FLOW_CREATION);
+                }
+            } else if (!flowCreated.containsKey(new ImmutablePair<>(bucketId, flowName))) {
+                createFlow(client, flowCreated, flowId, flowName, flowDescription, bucketId);
+            }
+
+            // Create missing flow versions
+            if (!versionMap.getOrDefault(flowId, Collections.emptyList()).contains(flowVersion)) {
+                //update storage location
+                final String registryUrl = getRequiredArg(properties, CommandOption.URL);
+
+                updateStorageLocation(snapshot.getFlowContents().getVersionedFlowCoordinates(), registryUrl);
+                for (VersionedProcessGroup processGroup : snapshot.getFlowContents().getProcessGroups()) {
+                    updateStorageLocation(processGroup.getVersionedFlowCoordinates(), registryUrl);
+                }
```

Comment:
Right now I think this will only handle the root group and first level groups, but there could be a child group under version control at any arbitrary level. 
```
Main PG
  - PG Level 1 (not under VC)
    - PG Level 2 (not under VC) 
      - PG Level 3 (Version Controlled) 
```